ui/src/app/applications/components/applications-list/applications-list.tsx

```diff
@@ -308,18 +308,18 @@ export const ApplicationsList = (props: RouteComponentProps<{}>) => {
     const loaderRef = React.useRef<DataLoader>();
     const {List, Summary, Tiles} = AppsListViewKey;
 
-    function refreshApp(appName: string) {
+    function refreshApp(appName: string, appNamespace: string) {
         // app refreshing might be done too quickly so that UI might miss it due to event batching
         // add refreshing annotation in the UI to improve user experience
         if (loaderRef.current) {
             const applications = loaderRef.current.getData() as models.Application[];
-            const app = applications.find(item => item.metadata.name === appName);
+            const app = applications.find(item => item.metadata.name === appName && item.metadata.namespace === appNamespace);
             if (app) {
                 AppUtils.setAppRefreshing(app);
                 loaderRef.current.setData(applications);
             }
         }
-        services.applications.get(appName, 'normal');
+        services.applications.get(appName, appNamespace, 'normal');
```

Comment:
Interesting. I'd have thought the TypeScript build would have failed.

Reply:
Yes, that's what you and me would expect :). But alas, the build runs just fine on those things :thinking:

Reply:
I think no typescript issue here. 3rd argument is optional in `services.applications.get`.